### PR TITLE
frontend: Refactor shortcuts to use ShortcutTable, add shortcut

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Refactored YAML shortcuts to use new shared ShortcutTable component and added save shortcut from Sam's recent PR. Made small tweak to ShortcutTable to accommodate Mac Ctrl keys.

Screen Shot 2019-11-21 at 10 29 28 AM

Heads up at @openshift/team-ux-review.

### CODE REVIEW

File: frontend/public/components/edit-yaml.jsx [hunk 1]
```diff
@@ -709,13 +709,20 @@ const EditYAML_ = connect(stateToProps)(
...
-                            <ul>
-                              <li>Use Ctrl + Space to activate auto complete.</li>
-                              <li>
-                                Use {isMac ? 'Command' : 'Ctrl'} + Shift + O for document outlining.
-                              </li>
-                              <li>Hover over a property to view a description.</li>
-                            </ul>
+                            <ShortcutTable>
+                              <Shortcut macCtrl keyName="space">
+                                Use Ctrl + Space to activate auto complete.
```
Comment (spadgett) [inline frontend/public/components/edit-yaml.jsx]: I wouldn't repeat the key shortcuts in the message.

```suggestion
                                activate auto complete
```

And similar for the shortcuts below.

File: frontend/public/components/edit-yaml.jsx [hunk 2]
```diff
@@ -709,13 +709,20 @@ const EditYAML_ = connect(stateToProps)(
...
+                              </Shortcut>
+                              <Shortcut ctrl shift keyName="o">
+                                Use Shift + {isMac ? 'Command' : 'Ctrl'} + O for document outlining.
```
Comment (spadgett) [inline frontend/public/components/edit-yaml.jsx]:
```suggestion
                                view document outline
```

File: frontend/public/components/edit-yaml.jsx [hunk 3]
```diff
@@ -709,13 +709,20 @@ const EditYAML_ = connect(stateToProps)(
...
+                              </Shortcut>
+                              <Shortcut hover>
+                                Hover over a property to view a description.
```
Comment (spadgett) [inline frontend/public/components/edit-yaml.jsx]:
```suggestion
                                view property descriptions
```

File: frontend/public/components/edit-yaml.jsx [hunk 4]
```diff
@@ -709,13 +709,20 @@ const EditYAML_ = connect(stateToProps)(
...
+                              </Shortcut>
+                              <Shortcut ctrl keyName="s">
+                                Use {isMac ? 'Command' : 'Ctrl'} + S to save.
```
Comment (spadgett) [inline frontend/public/components/edit-yaml.jsx]:
```suggestion
                                save
```

File: frontend/public/components/edit-yaml.jsx (hunk 2 above)
Comment (christianvogt) [reply frontend/public/components/edit-yaml.jsx]: UX strings provided for dev-console topology began with upper case.
`View document outline`

**Comment by christianvogt** [comment]
@rebeccaalpert I created a PR with an update to `Shortcut` to better support the mac variants and to update the display of key names like `SPACE` => `Space`
see here: https://github.com/rebeccaalpert/console/pull/3

**Comment by rebeccaalpert** [comment]
Updated text:

Screen Shot 2019-11-21 at 10 29 28 AM

**Comment by spadgett** [comment]
We probably want to make the popover link "View shortcuts" to align with the dev console. We should also make sure we use consistent case in both places

**Comment by spadgett** [comment]
Disregard, it looks like dev console is now using just "Shortcuts"

**Comment by christianvogt** [comment]
@spadgett i suggested the change to `Shortcuts` because that's what the yaml editor had lol

**Pull request merged**